### PR TITLE
Added  dynamic component generation

### PR DESCRIPTION
**Pull request by Revolyssup**
**Description**

This PR fixes #

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
@Revolyssup @Shreyas220 will you dig in here?

@Aisuko the bottom of this design spec has more details - https://docs.google.com/document/d/1b8JAMzr3Rntu7CudRaYv6r6ccACJONAB5t7ISCaPNuA/edit#

**Comment by Revolyssup** [comment]
> @Revolyssup @Shreyas220 will you dig in here?
> 
> @Aisuko the bottom of this design spec has more details - https://docs.google.com/document/d/1b8JAMzr3Rntu7CudRaYv6r6ccACJONAB5t7ISCaPNuA/edit#

I think this PR is good to go from my side.(Except for one last quick change after the new meshkit release comes out)

**Comment by leecalcote** [comment]
New MeshKit released.

**Comment by leecalcote** [comment]
@Revolyssup, a build check failure - 

```
Error: ../../../go/pkg/mod/github.com/layer5io/meshkit@v0.2.33/utils/walker/git.go:6:2: package io/fs is not in GOROOT (/opt/hostedtoolcache/go/1.15.15/x64/src/io/fs)
```

**Comment by Revolyssup** [comment]
> @Revolyssup, a build check failure -
> 
> ```
> Error: ../../../go/pkg/mod/github.com/layer5io/meshkit@v0.2.33/utils/walker/git.go:6:2: package io/fs is not in GOROOT (/opt/hostedtoolcache/go/1.15.15/x64/src/io/fs)
> ```

the meshkit function is causing this. this is due to the workflow using older go version. io/fs requires 1.16 onwards.
@DelusionalOptimist

File: internal/config/config.go [hunk 1]
```diff
@@ -124,3 +125,17 @@ func NewKubeconfigBuilder(provider string) (config.Handler, error) {
 func RootPath() string {
 	return configRootPath
 }
+
+// GetFileNames takes the url of a github repo and the path to a directory. Then returns all the filenames from that directory
+func GetFileNames(owner string, repo string, path string) ([]string, error) {
+	g := walker.NewGit()
+	var filenames []string
+	err := g.Owner(owner).Repo(repo).Root(path).RegisterFileInterceptor(func(f walker.File) error {
+		filenames = append(filenames, f.Name)
```
Comment (utkarsh-pro) [inline internal/config/config.go:134]: @Revolyssup same issue here?

Comment (Revolyssup) [reply internal/config/config.go]: made the change

**Comment by DelusionalOptimist** [comment]
> the meshkit function is causing this. this is due to the workflow using older go version. io/fs requires 1.16 onwards.
@DelusionalOptimist

@Revolyssup done :+1:

**Comment by leecalcote** [comment]
Kicking the build...

**Comment by DelusionalOptimist** [comment]
Almost all green.
@Revolyssup can you look into the security check? We either need to use a safer method or put a nosec to ignore this vulnerability.
https://github.com/meshery/meshery-linkerd/runs/4120663421?check_suite_focus=true#step:4:41

Also, as you update meshkit version to v0.2.33 in other adapters in similar PRs , you might need to make similar changes to [`6cdef4a` (#241)](https://github.com/meshery/meshery-linkerd/pull/241/commits/6cdef4a04965ce2fbcab6d7e539522e0b317eee2) and [`1c33c49` (#241)](https://github.com/meshery/meshery-linkerd/pull/241/commits/1c33c4993b6160057428cee05da47c7ca3bbf5dc)

**Comment by Revolyssup** [comment]
@leecalcote @DelusionalOptimist  all green

File: main.go [hunk 2]
```diff
@@ -157,3 +159,67 @@ func registerCapabilities(port string, log logger.Handler) {
...
+	var crds []string
+	for _, n := range names {
+		if strings.HasSuffix(n, "-crd.yaml") {
+			crds = append(crds, n)
+		}
+	}
+
+	rel, err := config.GetLatestReleases(1)
+	if err != nil {
+		log.Info("Could not get latest version ", err.Error())
```
Comment (utkarsh-pro) [inline main.go:190]: Error log maybe?

**Review by utkarsh-pro — APPROVED** [review]
LGTM

**Pull request merged**